tests/test_client.py

```diff
@@ -355,7 +355,7 @@ def test_no_retry_create_network(self):
         client = self.get_client_with_retry()
         # Creating network doesn't retry, we directly get a 502
         t = time.time()
-        with pytest.raises(BadStatus) as exc:
+        with pytest.raises(ServerError) as exc:
```

Comment:
maybe test for 4xx too? with `httpretty` it should be easy. (and it could also test that we don't retry for 4xx either).

Reply:
I added a test (without httpretty), just did an invalid inference request